### ISSUE TITLE
Do not install tests

### ISSUE THREAD

**Issue by moubctez** [python-hyper/wsproto#117]
Tests should not be installed nor packaged.
Here is a patch:

```
--- setup.py.orig	2018-09-23 11:26:23.000000000 +0000
+++ setup.py
@@ -35,7 +35,7 @@ setup(
     author='Benno Rice',
     author_email='benno@jeamland.net',
     url='https://github.com/python-hyper/wsproto/',
-    packages=find_packages(),
+    packages=find_packages(exclude=['test']),
     package_data={'': ['LICENSE', 'README.rst']},
     package_dir={'wsproto': 'wsproto'},
     include_package_data=True,
```

**Comment by pgjones** [comment]
`Tests should not be installed nor packaged.`, from a brief google I find that the prevailing advice is to install the tests, could you point me at the source of your view?

**Comment by njsmith** [comment]
They should definitely be included in the sdist. Whether they should be included as part of the main package and installed along with main package is significantly more controversial – there are major projects that do it both ways. This is closely related to the question of where you put your tests: https://docs.pytest.org/en/latest/goodpractices.html#choosing-a-test-layout-import-rules

**Comment by njsmith** [comment]
Oh, something I hadn't noticed before is that wsproto is actually installing a top-level package named `test`. That's definitely not right. Generally the options are to keep your tests outside your main package (`test/` and `wsproto/` next to each other) and don't install the tests, or else keep your tests inside your main package (`wsproto/test/`) and install them. The way wsproto is doing it now pollutes the top-level package namespace.

**Comment by tharvik** [comment]
@njsmith is right, it is indeed polluting the top-level namespace, and also, it's unimportable as `test` is already an internal CPython module.
Currently, it's not installable on Gentoo without a patch as `test` is [flagged as a forbidden package name](https://github.com/gentoo/gentoo/blob/master/eclass/distutils-r1.eclass#L593) during installation.

**Comment by pgjones** [comment]
I've released this in 0.14.1 and it looks to be fixed, please reopen if it isn't.
